CAMEL-21858: Checking the httpMethod parameter when decorating the sp…

lsergio opened the pull request:
…an for http components

# Description

The code checks if the endpoint uri for http components has the httpMethod parameter. If so, this is used as the http method in the tracing span. If not, we keep the current logic.

# Target

- [X ] I checked that the commit is targeting the correct branch (Camel 4 uses the `main` branch)

# Tracking
- [X ] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

<!--
# *Note*: trivial changes like, typos, minor documentation fixes and other small items do not require a JIRA issue. In this case your pull request should address just this issue, without pulling in other changes.
-->

# Apache Camel coding standards and style

- [ X] I checked that each commit in the pull request has a meaningful subject line and body.

<!--
If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
-->

- [ X] I have run `mvn clean install -DskipTests` locally from root folder and I have committed all auto-generated changes.

<!--
You can run the aforementioned command in your module so that the build auto-formats your code. This will also be verified as part of the checks and your PR may be rejected if if there are uncommited changes after running `mvn clean install -DskipTests`.

You can learn more about the contribution guidelines at https://github.com/apache/camel/blob/main/CONTRIBUTING.md
-->

squakez reviewed (commented):
Thanks for the PR. I think the fix could be fine, but I'd avoid to apply it on the `AbstractHttpSpanDecorator` since this is something specific to how Camel manage the `camel-http` instead. IMO, the abstraction is fine as it is, as the idea of having a request parameter for http method is not something part of the http specification, but it is specific to `camel-http` component instead. If you move the logic into the `HttpSpanDecorator` I think it would be good though.

lsergio commented:
Thanks for the feedback, @squakez . I implemented the change in the camel-telemetry component as well, and also moved the httpMethod handling  to the HttpSpanDecorator.

davsclaus commented:
Can you add this for camel-vertx-http which has similar `httpMethod` option and same logic for determine method that was used

lsergio commented:
@davsclaus done!

lsergio commented:
I see the builds have failed, but it does not seem to be related to my changes. I rebased my branch, just in case.

davsclaus commented:
yeah it seems the build tests passed but somehow the job returned an error code in the shell.

davsclaus commented:
ah it was a doc build
